### PR TITLE
Update to null safety in 2.12

### PR DESCRIPTION
**Pull request by mit-mit**
Also fix a few lints from pedantic

### CODE REVIEW

**Review by dcharkes — APPROVED** [review]
lgtm

**Comment by mit-mit** [comment]
@jakemac53 also, if the changes looks good to you, we could publish this, right?

**Comment by dcharkes** [comment]
Note that we still might deprecate some of the things of this package and move them to `dart:ffi`, but we can always release a 2.something if that were to be the case.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## 1.0.0-nullsafety
```
Comment (jakemac53) [inline CHANGELOG.md:3]:
```suggestion
## 1.0.0-nullsafety.0
```

File: pubspec.yaml [hunk 2]
```diff
@@ -1,16 +1,16 @@
 name: ffi
-version: 0.2.0
+version: 1.0.0-nullsafety
```
Comment (jakemac53) [inline pubspec.yaml:2]:
```suggestion
version: 1.0.0-nullsafety.0
```

File: pubspec.yaml [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 
 dev_dependencies:
-  pedantic: ^1.0.0
-  test: ^1.6.8
+  pedantic: ^1.10.0-nullsafety
+  test: ^1.16.0-nullsafety
 
-# Cannot be published until NNBD is released to stable.
+# Cannot be published until null safety is released to beta.
 publish_to: none
```
Comment (jakemac53) [inline pubspec.yaml:16]:
```suggestion
```

File: CHANGELOG.md (hunk 1 above)
Comment (mit-mit) [reply CHANGELOG.md:3]: Lol, I violated the rule I said earlier this week we should always follow :-)

**Comment by mit-mit** [comment]
@jakemac53 this is fine to publish now, right?

**Comment by mit-mit** [comment]
@dcharkes as you requested, I've changed this to `0.2.0-nullsafety.0`. @jakemac53 can you LGTM this version?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## 0.2.0-nullsafety.0
+
+Pre-release (non-stable) release supporting null safety.
+Only works with Dart 2.12.0 releases.
```
Comment (jakemac53) [inline CHANGELOG.md:6]:
```suggestion
Requires Dart 2.12.0 or greater.
```

**Review by jakemac53 — APPROVED** [review]

**Comment by mit-mit** [comment]
@jakemac53 this can be published, right?

**Pull request merged**